manifest/morph/morph.go

```diff
@@ -11,8 +11,16 @@ import (
 )
 
 // ValueToType transforms a value along a new type and returns a new value conforming to the given type
-func ValueToType(v tftypes.Value, t tftypes.Type, p *tftypes.AttributePath) (tftypes.Value, []*tfprotov5.Diagnostic) {
-	var diags []*tfprotov5.Diagnostic
+func ValueToType(v tftypes.Value, t tftypes.Type, p *tftypes.AttributePath) (val tftypes.Value, diags []*tfprotov5.Diagnostic) {
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
I was having a look at the godoc for tftypes and found this comment:

```
If the passed value is not a valid value for the passed type, NewValue will panic. Any value and type 
combination that does not return an error from ValidateValue is guaranteed to not panic. When calling 
NewValue with user input with a type not known at compile time, it is recommended to call ValidateValue 
before calling NewValue, to allow graceful handling of the error.
```

So it looks like although `NewValue()` will produce panics, it is guaranteed not to panic if we use `ValidateValue` with the same arguments first, which does return an error which we can then use to return a diagnostic message. This is slightly better than just capturing any panic produced by this function. 

It means we will need to look at each call site for `tftypes.NewValue()` and add a call to `ValidateValue()` and return a diag if it fails. Perhaps we can make a `newValue()` function to wrap this so we don't need to repeat ourselves a bunch of times.